### PR TITLE
Fixed more CheckedSession warning messages stacklevel and make test suite free of warning messages

### PR DESCRIPTION
**Pull request by gdementen**
@alixdamman: I would be grateful if you could try to keep the test suite free of (expected) warnings

### CODE REVIEW

**Comment by gdementen** [comment]
The "source"/stacklevel of warnings is now correct in all but one weird case (see the FIXME next to pandas_hdf).

Concerning the CheckedSession warnings, I simply marked all the warnings I received as expected. Maybe some of them were not. I didn't really analyze the code to determine that. Could you check if that seems alright to you?

Also, some tests (data) could be changed to avoid the warnings in the first place but I didn't make that exercice and I don't think it's worth it to do it now.

File: larray/core/checked.py [hunk 1]
```diff
@@ -383,13 +383,15 @@ class CheckedSession(Session, AbstractCheckedSession, metaclass=ModelMetaclass):
         dtype: int32
         memory used: 968 bytes
 
-        >>> # note that it still possible to add undeclared variables to a checked session
-        >>> # but this must be done with caution.
-        >>> m.undeclared_var = 'undeclared_var'
+        It is possible to add undeclared variables to a checked session
+        but this will print a warning:
+
+        >>> m.undeclared_var = 'my_value'                   # doctest: +SKIP
```
Comment (alixdamman) [inline larray/core/checked.py:389]: Is the warning tested in test_checkedsession.py ?

File: larray/core/checked.py [hunk 2]
```diff
@@ -501,11 +501,7 @@ def __setstate__(self, state: Dict[str, Any]) -> None:
             object.__setattr__(self, '__dict__', state['__dict__'])
 
         def dict(self, exclude: Set[str] = None):
-            d = dict(self.items())
-            for name in exclude:
-                if name in d:
-                    del d[name]
-            return d
+            return {k: v for k, v in self.items() if k not in exclude}
```
Comment (alixdamman) [inline larray/core/checked.py:508]: nice simplification

**Review by alixdamman — APPROVED** [review]
I am just asking 2 questions but LGTM.
Thanks for the cleanup of CheckedSession and of the tests associated with Session and CheckSession.

File: larray/core/checked.py (hunk 1 above)
Comment (gdementen) [reply larray/core/checked.py:389]: yes (cs.ax = 'ax')

File: setup.cfg [hunk 3]
```diff
@@ -31,6 +31,7 @@ addopts = -v --doctest-modules
           #--cov
 
 filterwarnings =
+    ignore::DeprecationWarning:flake8.*:
```
[the remark this answers is not in the document]
Comment (gdementen) [reply setup.cfg:34]: To ignore warnings coming from flake8 code itself: it uses some API which has been deprecated in recent versions of another package I am not sure of the name anymore (importlib-metadata or something like that)

**Pull request merged**